### DISCUSSION TITLE
Automatically populate author_id from auth.uid()

### DISCUSSION THREAD

**Discussion by aswinmohanme** [supabase/supabase#2812] [Questions] [answered]
I have a books table, that has an author_id column. I have enabled RLS for it. Is there any way to specify on the RLS statement that I want the author_id to be the auth.uid() on insert.

**Comment by soedirgo** [comment, marked as answer]
This is probably tangential to RLS @aswinmohanme, you can try setting the default value for `author_id` with the following SQL:
```sql
alter table books alter author_id set default auth.uid();
```
Note that the value will be `null` on non-signed in users.
